owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go

```diff
@@ -33,6 +33,29 @@ func ReadCookie(c echo.Context) error {
 	return c.String(http.StatusOK, "")
 }
 
+func GetCurrentUser(c echo.Context) (string, error) {
+	cookie, err := c.Cookie("sessionIDa5")
+	if err != nil {
+		return "", err
+	}
+
+	token, err := jwt.Parse(cookie.Value, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte("secret"), nil
```

Comment:
Ops, is that a hardcoded password on your code? 😅

Reply:
You tell me: https://github.com/globocom/secDevLabs/blob/master/owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/session.go#L73 :sweat_smile:

Reply:
🐰Easter Egg Found! 👏🏻